src/tlo/methods/bladder_cancer.py

```diff
@@ -673,43 +680,54 @@ def apply(self, person_id, squeeze_factor):
         if not pd.isnull(df.at[person_id, "bc_date_diagnosis"]):
             return hs.get_blank_appt_footprint()
 
-        # Use a cytoscope to diagnose whether the person has bladder Cancer:
-        dx_result = hs.dx_manager.run_dx_test(
-            dx_tests_to_run='cytoscopy_for_bladder_cancer_given_blood_urine',
-            hsi_event=self
-        )
-
-        if dx_result:
-            # record date of diagnosis:
-            df.at[person_id, 'bc_date_diagnosis'] = self.sim.date
-
-            # Check if is in metastatic:
-            in_metastatic = df.at[person_id, 'bc_status'] == 'metastatic'
-
-            # If diagnosis detects cancer, we assume classification as metastatic is accurate
-            if not in_metastatic:
-                # start treatment:
-                hs.schedule_hsi_event(
-                    hsi_event=HSI_BladderCancer_StartTreatment(
-                        module=self.module,
-                        person_id=person_id
-                    ),
-                    priority=0,
-                    topen=self.sim.date,
-                    tclose=None
-                )
+        # Check consumables are available
+        # TODO: replace with cystoscope
+        cons_avail = self.get_consumables(item_codes=self.module.item_codes_bladder_can['screening_biopsy_core'],
+                                          optional_item_codes=
+                                          self.module.item_codes_bladder_can['screening_biopsy_optional'])
+
+        if cons_avail:
+            # Use a biopsy to diagnose whether the person has bladder Cancer
+            # If consumables are available update the use of equipment and run the dx_test representing the biopsy
+            self.add_equipment({'Cystoscope', 'Ordinary Microscope'})
+
+            # Use a cystoscope to diagnose whether the person has bladder Cancer:
+            dx_result = hs.dx_manager.run_dx_test(
+                dx_tests_to_run='cystoscopy_for_bladder_cancer_given_blood_urine',
```

Comment:
Are the consumables, the availability of which is checked above the consumables used for the 'assess_hypertension' test? If so, shouldn't be their availability checked within the run_dx_test as it is done for other tests?

Reply:
I dont know if it matters a huge amount at this stage as there is some varation across the model - importantly the dx_test function doesnt allow for specification of core and optional consumables like the function `get_consumables` does...

We could raise an issue for this to be changed and for it to be done properly across the model.